flang/lib/Optimizer/OpenMP/DoConcurrentConversion.cpp

```diff
@@ -255,9 +254,21 @@ bool isIndVarUltimateOperand(mlir::Operation *op, fir::DoLoopOp doLoop) {
   return false;
 }
 
+/// For the \p doLoop parameter, find the operations that declares its induction
+/// variable or allocates memory for it.
+mlir::Operation *findLoopIndVarMemDecl(fir::DoLoopOp doLoop) {
+  mlir::Value result = nullptr;
+  mlir::visitUsedValuesDefinedAbove(
+      doLoop.getRegion(), [&](mlir::OpOperand *operand) {
+        if (isIndVarUltimateOperand(operand->getOwner(), doLoop))
+          result = operand->get();
```

Comment:
What is supposed to happen when multiple match? An early exit doesn't seem possible, but can you add an `assert(!result && "loop can have only one induction variable")`, respectively comment about that an arbitrary one is selected?

Reply:
Only one should match since `fir::DoLoopOp` is [defined](https://github.com/llvm/llvm-project/blob/main/flang/include/flang/Optimizer/Dialect/FIROps.td#L2175) as such.

Reply:
Done.